README.md

```diff
@@ -1 +1,186 @@
+
+- `template` or `templateUrl` - specifies the template that we use for that route (this works exactly like it would in a directive)
+- `controller` - specifies the controller that we use for that route (works the same as when we provide a string as the controller value in a directive)
+- `resolve` - an object of resolves that provides the data we need for that route (more on this later)
+
+Let's configure our `/user` route to use `user.html` as a template, and the `UserController` as its controller.
+
+```js
+angular
+	.module('app', ['ngRoute'])
+	.config(function ($routeProvider) {
+		$routeProvider
+			.when('/user', {
+				templateUrl: 'views/user.html',
+				controller: 'UserController'
+			});
+	});
+```
+
+Awesome! Now, when we go to `/#/user` in our application, we will see the user view, using the user controller to manage its data. Wait a minute - no we won't.
+
+In order to actually display our routes on the page, we need to tell `ngRoute` where to actually *put* the user view. We can do this using the directive `ng-view`.
+
+In our HTML, we would do something like this:
+
+```html
+<div ng-app="app" class="app">
+	<div ng-view></div>
+</div>
+```
+
+The `ng-view` directive will then put whatever route's template as it's contents when we go to that route - awesome!
+
+### Dynamic Views
+
+Being able to go to `/user` is great, but what if we wanted to a view a different user's profile? Well, we would need to specify the ID of that user. But our route isn't setup to receive an ID, and we can't hard code *every* ID into our router. This is where dynamic views come in.
+
+Much like when we used `$resource`, we can put variables into our URLs by using a colon. For instance, if we wanted to be able to go to `/user/1231` and `/user/94343` etc, we'd put `:id` into our URL.
+
+```js
+angular
+	.module('app', ['ngRoute'])
+	.config(function ($routeProvider) {
+		$routeProvider
+			.when('/user/:id', {
+				templateUrl: 'views/user.html',
+				controller: 'UserController'
+			});
+	});
+```
+
+This now allows us to go to the URLs we mentioned above. Awesome! But how do we actually know what the user's ID is? We'd need to go and fetch the user data so it's important that we can access what the variable is actually equal to.
+
+This is where another service provided by `ngRoute` comes in - `$routeParams`. We can inject this into our controllers and see what the ID variable is equal to.
+
+```js
+function UserController($routeParams) {
+	// $routeParams = { id: 1231 };
+}
+
+angular
+	.module('app')
+	.controller('UserController', UserController);
+```
+
+Now that we can access all of our route params, we can now go off and fetch the data for that user - simple!
+
+## Resolving Data
+
+Now, we could go and fetch user data in our controller, but that means that the view will load and then we will have a flicker between the view having no user data and the view being populated with data.
```

Comment:
This is an important concept, so explain this more slowly. Break down exactly what Angular is doing - first, the user is visiting the route - then the template is loading and the controller function is firing, then the data is loading. Therefore, the user will see the view without any data being populated first.